Sources/TuistCore/Cache/CacheDirectoriesProvider.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol CacheDirectoriesProviding: AnyObject {
```

Comment:
I'd differentiate between the caching feature of Tuist, and the internal caches that Tuist uses to speed up project generation. That said, I'd rename this to something more generic, `ProjectDirectoriesProviding` and `ProjectDirectoriesProvider`. Given a project configuration and its path, the instance returns paths to several directories.